ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -415,16 +431,41 @@ def _get_events_join_query(self) -> str:
                 AND event.event NOT IN funnel_step_names
         """
 
+    def _get_aggregation_join_query(self):
+        if self._filter.aggregation_group_type_index is None:
+            person_query, person_query_params = ClickhousePersonQuery(
+                self._filter, self._team.pk, ColumnOptimizer(self._filter, self._team.pk)
+            ).get_query()
+
+            return (
+                f"""
+                JOIN ({person_query}) person
+                    ON person.id = funnel_people.person_id
+            """,
+                person_query_params,
+            )
+        else:
+            return GroupsJoinQuery(
+                self._filter, self._team.pk, ColumnOptimizer(self._filter, self._team.pk)
+            ).get_join_query(["funnel_people.person_id"])
```

Comment:
Proposal: Let's move this argument to GroupsJoinQuery constructor and add excessive documentation on what/why this exists.

Constructor since that's more in-line with how PersonQuery works.

Reply:
There's only a single argument in this case, because we always expect to only have one group.

It made sense to me to keep an array with multiple possible options, since the join query can join multiple groups, and it wouldn't make sense to use the same alias to join on every group.

Reply:
The array logic however doesn't really work - again, it's way too coupled, etc.

Rather than try to future-proof it I'd go with the const approach for now. Then later as the need arises we can e.g. refactor it to be a lambda or something like that easily.